### DISCUSSION TITLE
How to rename rows? (e.g. variable names)

### DISCUSSION THREAD

**Discussion by lepromatous** [rstudio/gt#966] [Q&A]
Is there a way to rename the variable names in the gt output table?  For example, it would be awesome if you could just pass a vector of new variable names in an option to rename the rows.  I know whatever is in the data frame, name-wise, is what is used as the row labels in the gt() object, but variable names are usually not self explanatory or pretty, and renaming columns in a data frame to make them pretty is pretty much insanity.  Specific example: if my variable name is 'race_aa' and it is coded 1,0, where 1 is "Black/African American", and 0 is "Other", I don't want race_aa to show as the rows - I want "Black/African American" to show.  I would apply factor labels to get the 0/1 to show appropriately., however. Renaming my data frame with variable names with special characters or spaces is not cool. I have to be missing something.  Any ideas are much appreciated!

Something like:  df %>% gt() %>% append_row_names(new.names) where 'new.names' is a vector of names in the same order as what is in the output of gt()

**Comment by alex-lauer** [comment]
I don't quite know if I understand your problem correctly which is why it would be great to have a reprex. My 2 cents: You can use `dplyr::recode()` as part of your data prep prior to passing your tibble to gt. 
```
library(dplyr)
library(gt)

data("pizzaplace")
### We'll recode the size variable, which will be used as a rowname column

pizzaplace %>% 
  dplyr::filter(type %in% c("chicken", "classic", "veggie")) %>% 
  dplyr::select(name, size, type, price) %>% 
  head(n = 10) %>% 
  dplyr::mutate(size = dplyr::recode(size, "S" = "Appetizer", "M" = "Moderately hungry", "L" = "Hungry", "XL" = "Super Hungry", "XXL" = "Game Night with Friends"),
                size = factor(size, levels = as.character(size_nms))) %>%
  dplyr::arrange(size) %>% 
  gt(rowname_col = "size", groupname_col = "type") 
```
In a clinical data workflow, you can make use of named vectors as code lists:
```
### Create a vector of preferred terms
size_nms <- c("S" = "Appetizer", "M" = "Moderately hungry", "L" = "Hungry", "XL" = "Super Hungry", "XXL" = "Game Night with Friends")

pizzaplace %>% 
  dplyr::filter(type %in% c("chicken", "classic", "veggie")) %>% 
  dplyr::select(name, size, type, price) %>% 
  head(n = 10) %>% 
  dplyr::mutate(size = dplyr::recode(size, !!!size_nms),
                size = factor(size, levels = as.character(size_nms))) %>%
  dplyr::arrange(size) %>% 
  gt(rowname_col = "size", groupname_col = "type") 
```
